src/collection/class.js

```diff
@@ -34,14 +33,10 @@ let elesfn = ({
 
       // check if ele has classes outside of those passed
       if( !changedEle ){
-        let classes = Object.keys( eleClasses );
+        for( let eleCls of eleClasses ){
 
-        for( let i = 0; i < classes.length; i++ ){
-          let eleCls = classes[i];
-          let eleHasClass = eleClasses[ eleCls ];
-          let specdClass = classesMap[ eleCls ]; // i.e. this class is passed to the function
-
-          if( eleHasClass && !specdClass ){
+          let specdClass = classesMap.has(eleCls);
+          if( !specdClass ){
             changedEle = true;
             break;
           }
```

Comment:
It doesn't make sense to `util.copy()` a `Set`, since it's not a plain object (`{}`).  See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

One way to copy a set is `new Set( oldSet.values() )`, but you might want to experiment with different ways of copying if that's slow.